modules/external-load-balancer/main.tf

```diff
@@ -33,17 +33,162 @@ resource "google_compute_backend_service" "application" {
   timeout_sec = 10
 }
 
-resource "google_compute_url_map" "application" {
-  name = "${var.prefix}application"
+resource "google_compute_health_check" "install_dashboard" {
+  name = "${var.prefix}install-dashboard"
+
+  check_interval_sec = 5
+  description        = "The TFE install dashboard UI health check."
+  tcp_health_check {
+    port = var.vpc_install_dashboard_tcp_port
+  }
+  timeout_sec = 4
+}
+
+resource "google_compute_backend_service" "install_dashboard" {
+  health_checks = [google_compute_health_check.install_dashboard.self_link]
+  name          = "${var.prefix}install-dashboard"
+
+  backend {
+    group = var.primaries_instance_group_self_link
+
+    balancing_mode        = "RATE"
+    description           = "The TFE primaries."
+    max_rate_per_instance = 333
+  }
+  backend {
+    group = var.secondaries_instance_group_manager_instance_group
+
+    balancing_mode        = "RATE"
+    description           = "The TFE secondaries."
+    max_rate_per_instance = 333
+  }
+  port_name   = "install-dashboard"
+  protocol    = "HTTPS"
+  timeout_sec = 10
+}
 
+resource "google_compute_url_map" "main" {
   default_service = google_compute_backend_service.application.self_link
-  description     = "The TFE application URL map."
+  name            = "${var.prefix}all-requests"
+
+  description = "The URL map for all TFE requests."
+  host_rule {
+    hosts        = [var.dns_fqdn]
+    path_matcher = "allpaths"
+
+    description = "Match the fully-qualified domain name of TFE."
+  }
+  path_matcher {
+    default_service = google_compute_backend_service.application.self_link
+    name            = "allpaths"
+
+    description = "Redirect requests to the application or the install dashboard path."
+    # This rule covers paths which are unique to the install dashboard.
+    path_rule {
+      # FIXME: https://tfe.emp-gcp.ptfedev.com/api/v1/support returns 502
+      paths = [
+        "/api/v1",
+        "/api/v1/*",
+        "/auditlog",
+        "/auditlog/*",
+        "/authenticate",
+        "/authenticate/*",
+        "/ceph",
+        "/ceph/*",
+        "/cluster",
+        "/cluster/*",
+        "/console",
+        "/console/*",
+        "/dashboard",
+        "/dashboard/*",
+        "/license",
+        "/license/*",
+        "/premkit",
+        "/premkit/*",
+        "/releases",
+        "/releases/*",
+        "/settings",
+        "/settings/*",
+        "/support",
+        "/support/*",
```

Comment:
one question is how we maintain parity on this with application endpoints? do we have a swagger for atlas on the application endpoints as there's updates?

Reply:
The asset paths below this are also a bit much in terms of maintaining an ongoing list.

Reply:
With this approach, Atlas is the default service so we only need to redirect Replicated paths, which were discovered manually.

Reply:
This approach is definitely brittle; it's just the way to make it work immediately. It would be much easier if we could configure Replicated to not assume it is on the root of the domain, or if GCP supported load balancing HTTPS on a port other than 443.

Reply:
All these are only replicated endpoints? if that's the case then this description seems wrong https://github.com/hashicorp/terraform-google-terraform-enterprise/pull/62/files/1ee24a36030c6f970d84e3f9a18e080b23c12dae#diff-592f67edb38b1913afa732fb7b59aa0aR85

Reply:
So the assets portion is already broken on a new deploy. We will need to investigate an alternative approach.